packages/diracx-web-components/components/shared/DataTable.tsx

```diff
@@ -327,337 +358,346 @@ export function DataTable<T extends Record<string, unknown>>(
-      ...prevState,
-      sort: [{ parameter: property, direction: isAsc ? "desc" : "asc" }],
-    }));
-  };
-
-  // Manage selection
-  const handleSelectAllClick = (event: React.ChangeEvent<HTMLInputElement>) => {
-    if (event.target.checked) {
-      const newSelected = rows.map((n: T) => n[rowIdentifier] as number);
-      setSelected(newSelected);
-      return;
-    }
-    setSelected([]);
-  };
-
-  const handleClick = (event: React.MouseEvent<unknown>, id: number) => {
-    const selectedIndex = selected.indexOf(id);
-    let newSelected: readonly number[] = [];
-
-    if (selectedIndex === -1) {
-      newSelected = newSelected.concat(selected, id);
-    } else if (selectedIndex === 0) {
-      newSelected = newSelected.concat(selected.slice(1));
-    } else if (selectedIndex === selected.length - 1) {
-      newSelected = newSelected.concat(selected.slice(0, -1));
-    } else if (selectedIndex > 0) {
-      newSelected = newSelected.concat(
-        selected.slice(0, selectedIndex),
-        selected.slice(selectedIndex + 1),
-      );
-    }
-
-    setSelected(newSelected);
-  };
+  const handleRequestSort = React.useCallback(
+    (_event: React.MouseEvent<unknown>, property: string) => {
+      const isAsc =
+        searchBody.sort &&
+        searchBody.sort[0]?.parameter === property &&
+        searchBody.sort[0]?.direction === "asc";
+      setSearchBody((prevState: SearchBody) => ({
+        ...prevState,
+        sort: [{ parameter: property, direction: isAsc ? "desc" : "asc" }],
+      }));
+    },
+    [searchBody, setSearchBody],
+  );
 
   // Manage pagination
-  const handleChangePage = (event: unknown, newPage: number) => {
-    setPage(newPage);
-  };
-
-  const handleChangeRowsPerPage = (
-    event: React.ChangeEvent<HTMLInputElement>,
-  ) => {
-    setRowsPerPage(parseInt(event.target.value, 10));
-    setPage(0);
-  };
+  const handleChangePage = React.useCallback(
+    (_event: unknown, newPage: number) => {
+      table.setPageIndex(newPage);
+    },
+    [table],
+  );
 
-  const isSelected = (name: number) => selected.indexOf(name) !== -1;
+  const handleChangeRowsPerPage = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      table.setPageSize(Number(event.target.value));
+      table.setPageIndex(0);
+    },
+    [table],
+  );
 
   // Manage context menu
-  const handleContextMenu = (event: React.MouseEvent, id: number) => {
-    event.preventDefault(); // Prevent default context menu
-    setContextMenu({
-      mouseX: event.clientX - 2,
-      mouseY: event.clientY - 4,
-      id,
-    });
-  };
+  const handleContextMenu = React.useCallback(
+    (event: React.MouseEvent, id: number) => {
+      event.preventDefault();
+      setContextMenu({
+        mouseX: event.clientX - 2,
+        mouseY: event.clientY - 4,
+        id,
+      });
+    },
+    [],
+  );
 
-  const handleCloseContextMenu = () => {
+  const handleCloseContextMenu = React.useCallback(() => {
```

Comment:
would it be better just to import `useCallback` ?  again just a style suggestion

Reply:
From what I understand (and please correct me if I am wrong), there is no clear advantage of one way against the other (`React.use...` vs `use...`) so it's mostly a question of personal preference.

Now it's true that we write both `React.use...` and `use...` throughout the code and it would be better to be consistent. Though, if we choose one way I think it would be better to enforce it with an automated check using ESLint for instance (because else I am pretty sure this will be forgotten in a few months/years).

I don't find any easy way to enforce that but I will better check, thanks!